frontend/src/actions/IndependentWidgetsActions.js

```diff
@@ -0,0 +1,259 @@
+          type: PATCH_ATTRIBUTES,
+          payload: { data: data[0].fieldsByName },
+        });
+      }
+    });
+  };
+}
+
+/*
+ * @method setViewAttributesData
+ * @summary change local value of an attribute's field
+ *
+ * @param {string} field - field name
+ * @param {any} value
+ */
+export function setViewAttributesData({ field, value }) {
+  return {
+    type: SET_ATTRIBUTES_DATA,
+    payload: { field, value },
+  };
+}
+
+/*
+ * @method deleteViewAttributes
+ * @summary delete attributes data
+ */
+export function deleteViewAttributes() {
+  return {
+    type: DELETE_ATTRIBUTES,
+  };
+}
+
+//
+// TABLE QUICK INPUT
+//
+
+/**
+ * @method deleteQuickInput
+ * @summary Remove quick input's data
+ */
+export function deleteQuickInput() {
+  return {
+    type: DELETE_QUICKINPUT,
+  };
+}
+
+/**
+ * @method setQuickinputPending
+ * @summary Set the `inProgress` flag of QuickInput to true
+ */
+export function setQuickinputPending() {
+  return {
+    type: PATCH_QUICKINPUT_PENDING,
+  };
+}
+
+/**
+ * @method setQuickinputDone
+ * @summary Set the `inProgress` flag of QuickInput to false
+ */
+export function setQuickinputDone() {
+  return {
+    type: PATCH_QUICKINPUT_DONE,
+  };
+}
+
+/**
+ * @method setQuickinputData
+ * @summary Save field changes locally
+ *
+ * @param {object} fieldData - object with field names as keys and their data as values
+ */
+export function setQuickinputData(fieldData) {
+  return {
+    type: SET_QUICKINPUT_DATA,
+    payload: { fieldData },
+  };
+}
+
+/*
+ * @method patchQuickInput
+ * @summary Save QuickInput changes on the server
+ *
+ * @param {number} windowId
+ * @param {string} docId
+ * @param {string} tabId
+ * @param {string} prop
+ * @param {any} value
+ */
+export function patchQuickInput({ windowId, docId, tabId, prop, value }) {
+  return (dispatch, getState) => {
+    const state = getState();
+    const id = state.widgetHandler.quickInput.id;
+
+    dispatch(setQuickinputPending());
+
+    return patchRequest({
+      entity: 'window',
+      docType: windowId,
+      docId,
+      tabId,
+      property: prop,
+      value,
+      subentity: 'quickInput',
+      subentityId: id,
+    }).then(({ data }) => {
+      const fields = data[0] && data[0].fieldsByName;
+
+      if (fields) {
+        dispatch(setQuickinputData(fields));
+      }
+
+      dispatch(setQuickinputDone());
+    });
+  };
+}
+/*
+ * @method fetchQuickInputData
+ * @summary Get data for QuickInput's fields
+ *
+ * @param {number} windowId
+ * @param {string} docId
+ * @param {string} tabId
+ */
+export function fetchQuickInputData({ windowId, docId, tabId }) {
+  return (dispatch) => {
+    return initQuickActions(
+      'window',
+      windowId,
+      docId,
+      tabId,
+      'quickInput'
+    ).then(({ data }) => {
+      dispatch({
+        type: FETCH_QUICKINPUT_DATA,
+        payload: {
+          data: parseToDisplay(data.fieldsByName),
+          id: data.id,
+        },
+      });
+    });
```

Comment:
also some sort of err handling. Push the message to the store to be rendered by the notif box

Reply:
It's already handled https://recordit.co/Iq2Zyl4D0y